ravenpy/models/emulators/hypr.py

```diff
@@ -0,0 +1,602 @@
+        #                               MAXBAS,
+          :Parameters,               TIME_CONC,
+          :Units     ,  none,                d,
+                     1, {par_x15},
+        :EndSubBasinProperties
+        """
+        self.config.rvh.set_tmpl(rvh_tmpl)
+
+        #########
+        # R V C #
+        #########
+
+        self.config.rvc.hru_states[1] = HRUState(soil0=10, soil1=10, depression=20)
+
+    def derived_parameters(self):
+        params = cast(HYPR.Params, self.config.rvp.params)
+        self.config.rvt.snow_correction = params.par_x21
+        self.config.rvh.set_extra_attributes(par_x15=params.par_x15)
+        self._monthly_average()
+
+    # TODO: Support index specification and unit changes.
+    def _monthly_average(self):
+
+        if (
+            self.config.rvi.evaporation == "PET_FROMMONTHLY"
+            or self.config.rvi.ow_evaporation == "PET_FROMMONTHLY"
+        ):
+            # If this fails, it's likely the input data is missing some necessary variables (e.g. evap).
+            tas_cmd = cast(BaseDataCommand, self.config.rvt._var_cmds["tas"])
+            tasmin_cmd = cast(BaseDataCommand, self.config.rvt._var_cmds["tasmin"])
+            tasmax_cmd = cast(BaseDataCommand, self.config.rvt._var_cmds["tasmax"])
+            evspsbl_cmd = cast(BaseDataCommand, self.config.rvt._var_cmds["evspsbl"])
+
+            if tas_cmd:
+                tas = xr.open_dataset(tas_cmd.file_name_nc)[tas_cmd.var_name_nc]
+            else:
+                tasmax = xr.open_dataset(tasmax_cmd.file_name_nc)[
+                    tasmax_cmd.var_name_nc
+                ]
+                tasmin = xr.open_dataset(tasmin_cmd.file_name_nc)[
+                    tasmin_cmd.var_name_nc
+                ]
+                tas = (tasmax + tasmin) / 2.0
+
+            if evspsbl_cmd:
+                evap = xr.open_dataset(evspsbl_cmd.file_name_nc)[
+                    evspsbl_cmd.var_name_nc
+                ]
+
+            mat = tas.groupby("time.month").mean().values
+            mae = evap.groupby("time.month").mean().values
+
+            self.config.rvt.monthly_ave_evaporation = tuple(mae)
+            self.config.rvt.monthly_ave_temperature = tuple(mat)
+
+
+class HYPR_OST(Ostrich, HYPR):
+    def __init__(self, *args, **kwds):
+        super().__init__(*args, **kwds)
+
+        self.config.update(
+            identifier="hypr-ost",
+            algorithm="DDS",
+            max_iterations=50,
+            suppress_output=True,
+        )
+
+        ####################
+        # R V P (OST TMPL) #
+        ####################
+
+        rvp_tmpl = """
+        # tied parameters:
+        # (it is important for OSTRICH to find every parameter place holder somewhere in this file)
+        # (without this "para_x05" and "para_x06" wouldn't be detectable)
+        #    para_pow_x5     = pow_x05     =  10^(para_x05)       = 10^par_x05
+        #    para_pow_x6     = pow_x06     =  10^(para_x06)       = 10^par_x06
+
```

Comment:
@cjauvin I was concerned that the recent adjustments you made would lead to a correct population of those commented lines. Hence, my suggestion was to modify the previous version of `derived_parameters`:

```
    def derived_parameters(self):
        params = cast(HYPR.Params, self.config.rvp.params)
        self.config.rvp.derived_params = HYPR.DerivedParams(
            POW_X05=params.par_x05,
            POW_X06=params.par_x06,
        )
```
to the following version (not sure if the syntax would work...):
```
    def derived_parameters(self):
        params = cast(HYPR.Params, self.config.rvp.params)
        self.config.rvp.derived_params = HYPR.DerivedParams(
            POW_X05=params.par_x05,
            POW_X06=params.par_x06,
            par_x05 = np.log10(params.par_x05),
            par_x06 = np.log10(params.par_x06),
        )
```

This way all four parameters would be known and contain the correct values. This is important since sometimes Ostrich will need both versions of the value (e.g., POW_x05 and par_x05). Not in this example but for other models...

I am just afraid that the current implementation would not be generalizable for the other 7 models.

Reply:
Hi @julemai! What you propose would certainly work for sure, but I have 3 questions:

1. Why should we be concerned about Ostrich-related stuff in the emulator (I'm talking about the specific `hypr.HYPR` class, which is Raven-only, as opposed to `hypr.HYPR_OST`, which is Raven+Ostrich) that is (to my knowledge) completely devoid of any Ostrich stuff? (this was the spirit of my last commit, to be clear)
2. Why do you say that this approach wouldn't generalize to the other models?
3. .. I forgot my third point..